### PR TITLE
Added ACL-Metrics

### PR DESCRIPTION
**Pull request by aradz44**
**Description**
Adding new ACL metrics to the server, that tracking :
- user invalid command access
- user invaild key access
- user authentication failure

The metrics will be shown when using the command INFO with sections ALL/ACL/""

### CODE REVIEW

**Comment by ranshid** [comment]
LGTM - can you please add some tests to verify the statistics are updated correctly?

**Comment by aradz44** [comment]
> 

sure

File: tests/unit/acl.tcl [hunk 1]
```diff
@@ -730,6 +730,36 @@ start_server {tags {"acl external:skip"}} {
...
+    } {*WRONGPASS*}
+
+    # If a user try to access an unauthorized  command the metric increases
+    test {ACL-Metrics invalid command accesses} {
+        set current_invalid_accesses [s invalid_cmd_accesses]
+        r ACL setuser invalidcmduser on >passwd nocommands
+        r AUTH invalidcmduser passwd
+        catch {r acl list} err
+        r AUTH default ""
+        assert {[s invalid_cmd_accesses] eq [expr $current_invalid_accesses + 1]}
```
Comment (ranshid) [inline tests/unit/acl.tcl]: maybe we want to make sure other stats are not changing?

File: tests/unit/acl.tcl [hunk 2]
```diff
@@ -730,6 +730,36 @@ start_server {tags {"acl external:skip"}} {
        catch {r ACL load} err
        set err
     } {*Redis instance is not configured to use an ACL file*}
+
+    # If there is an AUTH failure the matric increases
+    test {ACL-Metrics user AUTH failure} {
```
Comment (ranshid) [inline tests/unit/acl.tcl:756]: add test for hello command as well

File: tests/unit/acl.tcl [hunk 3]
```diff
@@ -730,6 +730,63 @@ start_server {tags {"acl external:skip"}} {
...
+        set current_invalid_key_accesses [s invalid_key_accesses]
+        catch {r AUTH notrealuser 1233456} err
+        assert {[s auth_failure_times] eq [expr $current_auth_failures + 1]}
+        assert {[s invalid_cmd_accesses] eq $current_invalid_cmd_accesses}
+        assert {[s invalid_key_accesses] eq $current_invalid_key_accesses}
+        assert_match "*WRONGPASS*" $err
+    }
+
+    # If there is an AUTH failure during HELLO command the matric increases
+    test {ACL-Metrics user AUTH failure} {
```
Comment (ranshid) [inline tests/unit/acl.tcl]: same test name?

Comment (aradz44) [reply tests/unit/acl.tcl]: ops

**Pull request closed without merging**